sympy/solvers/solveset.py

```diff
@@ -552,7 +553,14 @@ def _solveset(f, symbol, domain, _check=False):
 
     # assign the solvers to use
     solver = lambda f, x, domain=domain: _solveset(f, x, domain)
-    if domain.is_subset(S.Reals):
+    if domain is S.Integers:
+        # calling methods from diophantine.py
+        solution = diop_solve(orig_f)
```

Comment:
`solveset` can solve equations in which there is an integer solution that are beyond the scope of diophantine. Also, the solution returned from `diophantine` does not claim to be complete (and probably shouldn't be, e.g. in the case of sums of squares there is a lot of redundancy in returning the full solution). The solution from `solveset` should be complete. So if the solution can be be found -- there should be an error if it can't -- then the signs and orders of the results should be permuted as necessary. `classify_diop` could be used to know when to do that; but there is no current hardcoded method of indicating when only an unpermuted subset of solutions is being returned from diophantine.